src/dataflow-types/src/client/controller/compute.rs

```diff
@@ -312,32 +314,45 @@ impl<'a, C: Client<T>, T: Timestamp + Lattice> ComputeController<'a, C, T> {
 
         // // Validate that the ids exist.
         // self.validate_ids(frontiers.iter().map(|(id, _)| *id))?;
+        let policies = frontiers
+            .into_iter()
+            .map(|(id, frontier)| (id, ReadPolicy::ValidFrom(frontier)));
+        self.set_read_policy(policies.collect()).await;
+    }
 
-        let mut updates = BTreeMap::new();
-        for (id, mut frontier) in frontiers.into_iter() {
-            // If-let to evade some identifiers incorrectly sent to us.
+    pub async fn set_read_policy(&mut self, policies: Vec<(GlobalId, ReadPolicy<T>)>) {
```

Comment:
Mind adding a doc string on this one? Wasn't clear to me whether this wholesale replaced read policies, or just overwrote the read policies for the specified IDs.

Reply:
Will do.